Moq.Tests/SequenceExtensionsFixture.cs

```diff
@@ -153,13 +155,39 @@ public void Verify_can_verify_invocation_count_for_sequences()
 			mock.Verify(m => m.Do(), Times.Exactly(2));
 		}
 
+		[Fact]
+		public void Func_are_invoked_deferred()
+		{
+			var mock = new Mock<IFoo>();
+			var i = 0;
+			mock.SetupSequence(m => m.Do())
+				.Returns(() => i);
+
+			i++;
+
+			Assert.Equal(i, mock.Object.Do());
+		}
+
+		[Fact]
+		public void Func_can_be_treated_as_return_value()
+		{
+			var mock = new Mock<IFoo>();
+			Func<int> func = () => 1;
+			mock.SetupSequence(m => m.GetFunc())
+				.Returns(func);
+
+			Assert.Equal(func, mock.Object.GetFunc());
+		}
+
 		public interface IFoo
 		{
 			string Value { get; set; }
 
 			int Do();
 
 			Task<int> DoAsync();
+
+			Func<int> GetFunc();
```

Comment:
Suggestion: Add another test for a method with return type `object`. This might be relevant because when `TResult := object`, both overloads `Returns(object)` and `Returns(Func<object>)` become relevant. (See also long review comment below.)

Reply:
:+1:

Reply:
:+1: